common/lib/xmodule/xmodule/lti_2_util.py

```diff
@@ -231,7 +231,7 @@ def clear_user_module_score(self, user):
         Returns:
             nothing
         """
-        self.set_user_module_score(user, None, None)
+        self.set_user_module_score(user, 0, self.max_score())
```

Comment:
This is one of the dumbest and most confusing test failures I've seen in a while. This call is trying to set a score of `(None, None)`, and then after passing through the grade signal handlers, lti complains that the values of possible and earned cannot be None.

```
    And the LTI provider deletes my grade and feedback                                              # lms/djangoapps/courseware/features/lti.py:367
ERROR:courseware.module_render:HEY LOOK AT THIS
ERROR:courseware.module_render:['max_value', 'user_id', 'value']
ERROR:lms.djangoapps.grades.signals.handlers:EFISCHER DEBUG SPECIAL!
ERROR:lms.djangoapps.grades.signals.handlers:kwargs: {{'only_if_higher': None, 'user_id': 123, 'weight': 10.0, 'signal': <django.dispatch.dispatcher.Signal object at 0x7f319e9cbf50>, 'modified': datetime.datetime(2016, 12, 8, 21, 4, 38, 203510, tzinfo=<UTC>), 'course_id': u'edx/test_course_a/Test_Course', 'raw_possible': None, 'raw_earned': None, 'usage_id': u'i4x://edx/test_course_a/lti/LTI'}}
  File "/usr/lib/python2.7/threading.py", line 783, in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib/python2.7/SocketServer.py", line 599, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/servers/basehttp.py", line 102, in __init__
    super(WSGIRequestHandler, self).__init__(*args, **kwargs)
  File "/usr/lib/python2.7/SocketServer.py", line 655, in __init__
    self.handle()
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/lettuce/django/server.py", line 95, in handle
    handler.run(self.server.get_app())
  File "/usr/lib/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/staticfiles/handlers.py", line 63, in __call__
    return self.application(environ, start_response)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/wsgi.py", line 189, in __call__
    response = self.get_response(request)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 846, in handle_xblock_callback_noauth
    return _invoke_xblock_handler(request, course_id, usage_id, handler, suffix, course=course)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 984, in _invoke_xblock_handler
    resp = instance.handle(handler, req, suffix)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/xblock/mixins.py", line 86, in handle
    return self.runtime.handle(self, handler_name, request, suffix)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1334, in handle
    return super(MetricsMixin, self).handle(block, handler_name, request, suffix=suffix)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/xblock/runtime.py", line 1025, in handle
    results = handler(request, suffix)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/lti_2_util.py", line 85, in lti_2_0_result_rest_handler
    return self._lti_2_0_result_put_handler(request, real_user)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/lti_2_util.py", line 217, in _lti_2_0_result_put_handler
    self.clear_user_module_score(real_user)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/lti_2_util.py", line 234, in clear_user_module_score
    self.set_user_module_score(user, None, None)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/lti_2_util.py", line 263, in set_user_module_score
    'user_id': user.id,
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 477, in publish
    only_if_higher=event.get('only_if_higher'),
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/dispatch/dispatcher.py", line 189, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/signals/handlers.py", line 136, in score_published_handler
    modified=score_modified_time,
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/dispatch/dispatcher.py", line 189, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/signals/handlers.py", line 149, in problem_raw_score_changed_handler
    from traceback import print_stack; print_stack()
ERROR:edx.lti_provider:Outcome Service: Required signal parameter is None. points_possible: None, points_earned: None, user_id: 123, course_id: edx/test_course_a/Test_Course, usa
``